### ISSUE TITLE
RLS hangs during install

### ISSUE THREAD

**Issue by epigramengineer** [NixOS/nixpkgs#93499]
**Describe the bug**
`rls` hangs indefinitely when installing

**To Reproduce**
```
❯ nix-env -i rls
....
   Compiling cargo v0.45.1 (/build/rustc-1.44.1-src/src/tools/cargo)
    Finished release [optimized] target(s) in 7m 03s
/build/rustc-1.44.1-src
running tests
no Makefile or custom buildPhase, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/fa4hvsckbjq80dhck6lxmwgx3waxjck1-rls-1.44.1
shrinking /nix/store/fa4hvsckbjq80dhck6lxmwgx3waxjck1-rls-1.44.1/bin/.rls-wrapped
strip is /nix/store/h4v5qdxlmnh7xfpl7pwzrs8js7220bz2-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/fa4hvsckbjq80dhck6lxmwgx3waxjck1-rls-1.44.1/bin
patching script interpreter paths in /nix/store/fa4hvsckbjq80dhck6lxmwgx3waxjck1-rls-1.44.1
checking for references to /build/ in /nix/store/fa4hvsckbjq80dhck6lxmwgx3waxjck1-rls-1.44.1...
running install tests
/build/rustc-1.44.1-src/src/tools/rls /build/rustc-1.44.1-src
Running cargo test --release --target x86_64-unknown-linux-gnu --frozen --
    Blocking waiting for file lock on build directory
```

**Expected behavior**
Not to block waiting for lock on build directory indefinitely (30+ minutes).

**Notify maintainers**
@symphorien

**Metadata**
```
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"` - host os: `Linux 5.4.43, NixOS, 20.03.2015.e7752db2fb6 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.5`
 - channels(root): `"nixos-20.03.2015.e7752db2fb6"` - channels(<user>): `"nixpkgs-20.09pre235189.d7e20ee25ed"`
 - nixpkgs: `/home/<user>/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by symphorien** [comment]
I think this was broken by #91689

There is a mistake in the PR: it moves the checkPhase to installCheckPhase but still sets `doCheck = true;` instead of `doInstallCheck = true;`

If I fix this, 
```diff
diff --git a/pkgs/build-support/rust/default.nix b/pkgs/build-support/rust/default.nix
index 9e9f2cb4e3b..211bd0062a2 100644
--- a/pkgs/build-support/rust/default.nix
+++ b/pkgs/build-support/rust/default.nix
@@ -210,7 +210,7 @@ stdenv.mkDerivation (args // {
     ${stdenv.lib.optionalString (buildAndTestSubdir != null) "popd"}
   '');
 
-  doCheck = args.doCheck or true;
+  doInstallCheck = args.doCheck or true;
 
   strictDeps = true;
 
```
then nix-du also exhibits the same behavior as what you saw for rls.
I presume the problem is general, then. That's all I have time to investigate for now.

cc @flakebi @zowok

**Comment by Flakebi** [comment]
That should be fixed by #93128, which is merged into staging. Sorry for the breakage!

**Comment by flokli** [comment]
Let's close this, it's fixed in staging and should hit master once https://github.com/NixOS/nixpkgs/pull/93465 is merged.
